scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_model.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import (absolute_import, division, print_function)
+
+from six import iteritems
+
+import mantid.simpleapi as mantid
+
+
+class FFTWrapper(object):
+
+    """
+    A class to wrap the different parts
+    of the FFT and its preprocessing.
+    This keeps the main FFT class simple.
+    """
+
+    def __init__(self, FFT):
+        self.name = "FFT"
+        self.model = FFT
+
+    def cancel(self):
+        self.model.cancel()
+
+    def loadData(self, inputs):
+        """
+        store the data in the wrapper for later
+        """
+        if "phaseTable" in inputs:
+            self.phaseTable = inputs["phaseTable"]
+        else:
+            self.phaseTable = None
+        if "preRe" in inputs:
+            self.preRe = inputs["preRe"]
+        else:
+            self.preRe = None
+        if "preIm" in inputs:
+            self.preIm = inputs["preIm"]
+        else:
+            self.preIm = None
+        if "FFT" in inputs:
+            self.FFT = inputs["FFT"]
+        else:
+            self.FFT = None
+        self.model.setRun(inputs["Run"])
+
+    def execute(self):
+        """
+        runs the relevant parts of the FFT and the preprocessing
+        """
+        if self.phaseTable is not None:
+            if self.phaseTable["newTable"]:
+                self.model.makePhaseQuadTable(self.phaseTable)
+            self.model.PhaseQuad()
+
+        if self.preRe is not None:
+            self.model.preAlg(self.preRe)
+
+        if self.preIm is not None:
+            self.model.preAlg(self.preIm)
+
+        if self.FFT is not None:
+            self.model.FFTAlg(self.FFT)
+
+    def output(self):
+        return
+
+
+class FFTModel(object):
+
+    """
+    A simple class which executes
+    the relevant algorithms for
+    the analysis.
+    """
+
+    def __init__(self):
+        self.name = "FFT"
+        self.alg = None
+
+    def cancel(self):
+        if self.alg is not None:
+            self.alg.cancel()
+
+    def setRun(self, run):
+        self.runName = run
```

Comment:
Just a general question; is it normal to have member variables of the class defined within member functions? It makes it hard to understand what information the object carries around. There are also many occasions where a member variable is set to `None` if not in a list of inputs; would it be easier to just declare them all `None` in the `__init__`?

Reply:
This is important for how it works later on

Reply:
I tend to agree with Tom here. I think it's clearer to define all of the attributes used by the class, even if they are set to None, in the class `__init__` so you can see what data the class deals with.

Reply:
an easy way to do it without duplicating code would be to just call `loadData` at the end of init with an empty dictionary